### PR TITLE
Add Database abstraction and helpers for various data files.

### PR DESCRIPTION
**Pull request by blambeau**

### CODE REVIEW

**Comment by blambeau** [comment]
@felixyz how about this Database (read-only) abstraction ? I think it makes Bmg easier to use on files, excel, and sql database.

File: README.md [hunk 1]
```diff
@@ -117,33 +118,36 @@ Bmg.sequel(:suppliers, sequel_db)
...
 ```ruby
-roo_options = { skip: 1 }
-r = Bmg.excel("path/to/a/file.xls", roo_options)
+csv_options = { col_sep: ",", quote_char: '"' }
+r = Bmg.csv("path/to/a/file.csv", csv_options)
 ```
 
-Options are directly transmitted to `Roo::Spreadsheet.open`, check roo's
-documentation.
+Options are directly transmitted to `::CSV.new`, check Ruby's standard
```
Comment (felixyz) [inline README.md:149]: Possible improvement: add the `headers` flag.

https://ruby-doc.org/stdlib-2.6.1/libdoc/csv/rdoc/CSV.html#class-CSV-label-CSV+with+headers

File: README.md [hunk 2]
```diff
@@ -240,6 +257,52 @@ restrictions down the tree) by overriding the underscored version of operators
 Have a look at `Bmg::Algebra` for the protocol and `Bmg::Sql::Relation` for an
 example. Keep in touch with the team if you need some help.
 
+## The Database abstraction
+
+The previous section focused on obtaining *relations*. In pratice you frequently have a collection of related relations hence a *database*:
```
Comment (felixyz) [inline README.md:262]: typo: practice

suggestion: drop "related"

File: README.md [hunk 3]
```diff
@@ -240,6 +257,52 @@ restrictions down the tree) by overriding the underscored version of operators
...
+
+* A SQL database with multiple tables
+* A list of data files, all in the same folder
+* An excel file with various sheets
+
+Bmg supports a simple Datbabase abstraction that serves those relations "by name", in a simple way. A database can also be easily dumped back to a data folder of json or csv files, or as a simple xlsx files with multiple sheets.
```
Comment (felixyz) [inline README.md:268]: "as a simple" => "as simple"

File: README.md [hunk 4]
```diff
@@ -240,6 +257,52 @@ restrictions down the tree) by overriding the underscored version of operators
...
+### Connecting to a SQL Database
+
+For a SQL database, connected with Sequel:
+
+```
+db = Bmg::Database.sequel(Sequel.connect('...'))
+db.suppliers # yields a Bmg::Relation over the `suppliers` table
+```
+
+### Connecting to data files in the same folder
```
Comment (felixyz) [inline README.md:279]: Make explicit that the files can be in different file formats (if that's the case)

File: README.md [hunk 5]
```diff
@@ -240,6 +257,52 @@ restrictions down the tree) by overriding the underscored version of operators
...
+db.suppliers # yields a Bmg::Relation over the `suppliers.(json,csv,yml)` file
+```
+
+### Dumping a Database instance
+
+As a data folder:
+
+```
+db = Bmg::Database.sequel(Sequel.connect('...'))
+db.to_data_folder('path/to/folder', 'json')
```
Comment (felixyz) [inline README.md:295]: Very nitpicky, but to me using a string to specify the output format doesn't feel right. Would prefer a constant or possibly (although it's really no different) a symbol.

**Review by felixyz — COMMENTED** [review]
I haven't looked at the implementation, only the changelog and readme, but I think this is a great feature! The docs could be expanded but let's do that over in the docs repo :)

If you want more feedback on the implementation, I can get back to this later in the week.

**Pull request closed without merging**